osi_logicallane.proto

```diff
@@ -0,0 +1,502 @@
+    // OpenDRIVE), the points must be chosen in a way that the lateral distance
+    // to the ideal line does not exceed 5cm. As shown in the following image:
+    //
+    // \image html line_approximation_error.svg "Approximation error"
+    // Approximation error green line.
+    //
+    // The Z error (difference in Z height between boundary_line and the "real"
+    // line) must also not exceed 5cm.
+    //
+    // Note: if two lanes have different Z heights (e.g. a driving lane is
+    // beside a sidewalk, where the sidewalk is 10cm higher than the road),
+    // then these lanes cannot share a boundary, since their boundaries have
+    // different Z heights.
+    //
+    repeated LogicalBoundaryPoint boundary_line = 2;
+
+    // The reference line for this boundary.
+    //
+    // The reference line is used as a coordinate system for this boundary.
+    // All points of this LogicalLaneBoundary must have S coordinates in the
+    // range [sStart,sEnd].
+    //
+    // The reference line should roughly have the same shape as the boundary, so
+    // that S coordinates continually increase along the lane.
+    //
+    optional Identifier reference_line_id = 3;
+
+    // The classification of the lane boundary.
+    //
+    optional LaneBoundary.Classification classification = 4;
+
+    // Optional external reference to the lane boundary source.
+    //
+    // \note For OpenDRIVE, there is no direct possibility to reference the
+    //       RoadMark, as there is no unique identifier in this sub-object.
+    //
+    // \note For non-ASAM Standards, it is implementation-specific how
+    //       source_reference is resolved.
+    //
+    // \note The value has to be repeated because one object may be derived
+    //       from more than one origin source, for example, from a scenario file
+    //       and from sensors.
+    //
+    repeated ExternalReference source_reference = 5;
+
+    // A point on the boundary
+    //
+    message LogicalBoundaryPoint {
+
+        // The position of the \c LaneBoundaryPoint.
+        //
+        optional Vector3d position = 1;
+
+        // S position of the LaneBoundaryPoint, measured along the parent's
+        // reference_line_id.
+        //
+        optional double s_position = 2;
+
+        // T position of the LaneBoundaryPoint, measured along the parent's
+        // reference_line_id.
+        //
+        optional double t_position = 3;
+    }
+}
+
+//
+// \brief A logical lane in the road network.
+//
+// A logical lane is part of a road. Compared to a physical lane (OSI type
+// \c Lane), its existence doesn't hinge on the existence of road markings. So
+// e.g. a narrow urban road with two driving directions but no road markings
+// in-between would be presented as two LogicalLanes, but only one Lane. So one
+// Lane can consist of multiple LogicalLanes. E.g. on intersections, each
+// driving path is one LogicalLane, but the whole area is one \c Lane of type
+// \c #TYPE_INTERSECTION.
```

Comment:
I think here we might need an even more precise definition of what a logical lane really is / what its boundaries are when not limited by physical lane boundaries (e.g. how wide are the overlapping logical lanes in narrow urban roads?). Or write that this is free for implementations to decide. 

While I currently don't have a general approach, I think in intersections it would be quite easy to define (width of ingoing and outgoing lanes) and also for the case you sketch in line 330ff. (width of the preceding and succeeding lanes.

Reply:
Hmm, the original purpose of this pull request was to make as much map information as possible available to OSI. So I would rather not define this too strictly, but say something like "lanes from the underlying map should be represented as accurately as possible". I think it is OK to split or merge them like this:

```
 --------------          --------  ------
 --> lane1       ==>  lane 2  lane 3
 --------------          --------  ------
```

Splitting them like this doesn't change any semantics. But the widths etc. should not be changed.

Rationale: if we define separate rules, then it WILL diverge from OpenDRIVE, which means that again OpenDRIVE maps cannot be represented in OSI correctly, and implementations will have more effort to convert between different formats.